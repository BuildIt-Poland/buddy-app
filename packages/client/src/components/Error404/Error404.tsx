import React from 'react';
import ErrorPage from 'components/ErrorPage';
import ERROR_404_DICTIONARY from './dictionary';

const Error404: React.FC = () => {
  return (
    <ErrorPage
      data-testid='error-page-404'
      title={ERROR_404_DICTIONARY.TITLE}
      message={ERROR_404_DICTIONARY.MESSAGE}
    />
  );
};

export default Error404;
