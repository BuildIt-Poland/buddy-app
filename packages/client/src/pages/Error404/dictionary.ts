interface Error404Dictionary {
  TITLE: string;
  MESSAGE: string;
}

const ERROR_404_DICTIONARY: Error404Dictionary = {
  TITLE: '404 - Page not found',
  MESSAGE: `Sorry! We looked everywhere but we can't find the page you're looking for.`,
};

export default ERROR_404_DICTIONARY;
