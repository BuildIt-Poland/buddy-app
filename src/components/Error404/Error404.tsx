import React from 'react';
import ErrorPage from 'components/ErrorPage';

const Error404: React.FC = () => {
  return (
    <ErrorPage
      data-testid='error-page-404'
      title={'404 - Page not found'}
      message={`Sorry! We looked everywhere but we can't find the page you're looking for.`}
    />
  );
};

export default Error404;
