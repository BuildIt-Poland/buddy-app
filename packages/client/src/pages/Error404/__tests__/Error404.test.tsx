import React from 'react';
import { create } from 'react-test-renderer';

import Error404 from '../Error404';

jest.mock('atoms/ErrorPage', () => 'ErrorPage');

describe('Component - Error404', () => {
  test('renders correctly', () => {
    const component = create(<Error404 />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
