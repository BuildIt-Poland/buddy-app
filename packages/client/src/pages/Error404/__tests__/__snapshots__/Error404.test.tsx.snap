// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Error404 renders correctly 1`] = `
<ErrorPage
  data-testid="error-page-404"
  message="Sorry! We looked everywhere but we can't find the page you're looking for."
  title="404 - Page not found"
/>
`;
